common/src/main/java/net/neoforged/gradle/common/util/ClasspathUtils.java

```diff
@@ -0,0 +1,35 @@
+package net.neoforged.gradle.common.util;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.zip.ZipFile;
+
+/**
+ * Utility class for dealing with classpaths and their entries.
+ */
+public class ClasspathUtils {
+
+    private ClasspathUtils() {
+        throw new UnsupportedOperationException("This is a utility class and cannot be instantiated.");
+    }
+
+    public static boolean isClasspathEntry(File entry) {
+        return entry.getName().endsWith(".jar") || entry.getName().endsWith(".zip");
+    }
+
+    public static boolean isMinecraftClasspathEntry(File entry) {
+        if (!isClasspathEntry(entry)) {
+            return false;
+        }
+
+        //Check if the file contains the class:
+        //net.minecraft.client.Minecraft
+        //This is a class that is always present in the Minecraft jar.
```

Comment:
Technically, this class is always present for the Minecraft _client_ JAR -- it won't be for the dedicated server JAR.

Would this cause problems in the future for a potential split-source system (where a source exists that's only the dedicated server)? Perhaps we should check for another prominent class that's likely to exist forevermore, such as `MinecraftServer` or `SharedConstants`?

Reply:
Sadly, that is the class FML checks itself for: https://github.com/neoforged/FancyModLoader/blob/89a32f970ba9137cb74e681af9cdaed0fc3e2085/loader/src/main/java/net/neoforged/fml/loading/targets/CommonUserdevLaunchHandler.java#L24

Reply:
Ah. Then could you please add a comment linking to that, so we know in the future to keep the two in sync?